js-pkg/react/src/main.tsx

```diff
@@ -250,38 +241,15 @@ export function YDocProvider(props: YDocProviderProps) {
   const [ctx, setCtx] = useState<YjsContextType | null>(null)
 
   useEffect(() => {
-    let canceled = false
-    let provider: YSweetProvider | null = null
     const doc = new Y.Doc()
+    const provider = createYjsProvider(doc, docId, authEndpoint, {
+      initialClientToken,
+      offlineSupport: props.offlineSupport,
+    })
 
-    ;(async () => {
-      provider = await createYjsProvider(doc, docId, authEndpoint, {
-        initialClientToken,
-        offlineSupport: props.offlineSupport,
-      })
-
-      if ((props.showDebuggerLink ?? true) && provider.clientToken) {
-        const url = debuggerUrl(provider.clientToken)
-        console.log(
-          `%cOpen this in Y-Sweet Debugger ⮕ ${url}`,
-          'font-size: 1.5em; display: block; padding: 10px;',
-        )
-        console.log(
-          '%cTo hide the debugger link, pass showDebuggerLink={false} to YDocProvider',
-          'font-style: italic;',
-        )
-      }
-
-      if (canceled) {
-        provider.destroy()
-        return
-      }
-
-      setCtx({ doc, provider })
-    })()
+    setCtx({ doc, provider })
 
     return () => {
-      canceled = true
       provider?.destroy()
```

Comment:
Nit: now that all the logic here is synchronous, we can drop the `?` after `provider`.